cmd/auth-rpc-client.go

```diff
@@ -146,21 +142,34 @@ func (authClient *AuthRPCClient) Call(serviceMethod string, args interface {
 	SetToken(token string)
 	SetTimestamp(tstamp time.Time)
 }, reply interface{}) (err error) {
-	// On successful login, attempt the call.
-	if err = authClient.Login(); err == nil {
-		// Set token and timestamp before the rpc call.
-		args.SetToken(authClient.serverToken)
-		args.SetTimestamp(time.Now().UTC())
-
-		// Call the underlying rpc.
-		err = authClient.rpc.Call(serviceMethod, args, reply)
-
-		// Invalidate token, and mark it for re-login on subsequent reconnect.
-		if err == rpc.ErrShutdown {
-			authClient.mu.Lock()
-			authClient.isLoggedIn = false
-			authClient.mu.Unlock()
+	loginAndCallFn := func() error {
+		// On successful login, proceed to attempt the requested service method.
+		if err = authClient.Login(); err == nil {
+			// Set token and timestamp before the rpc call.
+			args.SetToken(authClient.serverToken)
+			args.SetTimestamp(time.Now().UTC())
+
+			// Finally make the network call using net/rpc client.
+			err = authClient.rpc.Call(serviceMethod, args, reply)
 		}
+		return err
+	}
```

Comment:
Can you share snippet of code i don't see it anywhere.. 

- https://github.com/docker/docker/blob/master/cmd/dockerd/daemon.go#L176
- https://github.com/docker/docker/blob/master/cmd/dockerd/daemon.go#L88
- https://github.com/docker/docker/blob/master/cmd/dockerd/daemon.go#L258

if it is done somewhere it is not consistent, in-fact docker also lacks comments too so i wouldn't buy into it too much into it.

The style itself is fine, but it is not a sacrosanct necessity if Go upstream project can live with it i don't see why we would have any issues and we don't even have 1/10th of the contributors they have.